### ISSUE TITLE
JWT header and payload decode (base64 with dots)

### ISSUE THREAD

**Issue by alright21** [Ciphey/Ciphey#489]
### Problem:
There is no way to decode faster JWTs

### Solution:
They are base64 divided by a dot, there should be a way to check this, and it should be really easy to decode it

**Comment by SkeletalDemise** [comment]
Hello. I am wondering, what do you think we should do with the JWT signature? Each JWT (JSON Web Token) is split into 3 sections, the header, the payload, and the signature. Both the header and the payload can be decoded with Base64, but the signature can only be decrypted if you have the secret key. 

I've made a quick decoder that appends the Base64 encoded signature at the end. What do you think of this format?
`{"alg":"HS256","typ":"JWT"} {"userId":"110095","name":"John
Doe","roles":,"iat":1523817999,"exp":1524422799} Secret: rrpyIW0ftJGna1c_SHH6Hus8hBd2-CUqofaSIUHj0C0`

**Comment by alright21** [comment]
Hi again! Yes, that's what I was thinking about. I know that the signature cannot be decrypted, but the first two parts can. So if it is possible to recognize the dots separator and decrypt the header and the payload automatically, it will be great! The format is correct. Thank you for your answer!

**Comment by Cyclic3** [comment]
This would be a good use-case for the polymorphic decoder/checker, where we could create a type for JWTs, and set them as valid plaintexts

**Comment by SkeletalDemise** [comment]
That sounds good. I've already made a JWT decoder, but it doesn't work because the checkers don't consider it to be valid English or JSON.

**Comment by bee-san** [comment]
> That sounds good. I've already made a JWT decoder, but it doesn't work because the checkers don't consider it to be valid English or JSON.

Is this solved by Chi-Squared / Entropy, or does it need another checker?

**Comment by SkeletalDemise** [comment]
> Is this solved by Chi-Squared / Entropy, or does it need another checker?

No, the Chi Squared and/or Entropy checkers won't work. Cyclic's idea would work though.
